test/test_warning.rb

```diff
@@ -46,6 +46,90 @@ def teardown
     Warning.clear
   end
 
+  def test_warning_clear_ignore
+    Warning.ignore(/.*/)
+
+    assert_warning '' do
```

Comment:
To reduce horizontal space we could use:

```suggestion
    assert_warning('') { Warning.warn 'foo' }
```

Not sure if this is compatible with the style of this gem though :shrug:

Reply:
I think this is fine how it is.